uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/AgentControlImpl.java

```diff
@@ -141,11 +138,22 @@ public String getAgentId() {
     }
 
     @Override
-    public void shutdownAgent(String reason) {
+    public ConnectionControl getConnectionControl(@NonNull String connectionName) {
+        for (ConcurrentHashMap.Entry<Integer, ConnectionControlImpl> entry : connections.entrySet()) {
+            ConnectionControlImpl connectionControl = entry.getValue();
+            if (connectionName.equals(connectionControl.getConnectionName())) {
+                return connectionControl;
+            }
+        }
+        return null;
+    }
```

Comment:
Could we use Stream API? This will help to use parallel streams.

`        return connections.values().parallelStream()
                .filter(connectionControl -> connectionName.equals(connectionControl.getConnectionName()))
                .findAny().orElse(null);`

Reply:
In that case it useless to use parallelStream().